.github/workflows/python-test-coverage.yml

```diff
@@ -1,55 +1,69 @@
 name: Python Test Coverage
 
 on:
-  pull_request_target:
+  pull_request:
     branches: ["main", "feature*"]
     paths:
-      - "python/**"
-  workflow_run:
-    workflows: ["Python Unit Tests"]
-    types: 
-      - in_progress
+      - "python/semantic_kernel/**"
+      - "python/tests/unit/**"
+env:
+  # Configure a constant location for the uv cache
+  UV_CACHE_DIR: /tmp/.uv-cache
+
+permissions:
+  contents: write
+  checks: write
+  pull-requests: write
 
 jobs:
   python-tests-coverage:
     runs-on: ubuntu-latest
-    continue-on-error: true
-    permissions:
-      pull-requests: write
-      contents: read
-      actions: read
+    continue-on-error: false
+    defaults:
+      run:
+        working-directory: python
+    env:
+      UV_PYTHON: "3.10"
     steps:
-      - name: Wait for unit tests to succeed
-        uses: lewagon/wait-on-check-action@v1.3.4
-        with:
-          ref: ${{ github.event.pull_request.head.sha }}
-          check-name: 'Python Test Coverage'
-          repo-token: ${{ secrets.GH_ACTIONS_PR_WRITE }}
-          wait-interval: 90
-          allowed-conclusions: success
       - uses: actions/checkout@v4
       - name: Setup filename variables
         run: echo "FILE_ID=${{ github.event.number }}" >> $GITHUB_ENV
-      - name: Download Files
-        uses: actions/download-artifact@v4
+      - name: Set up uv
+        uses: astral-sh/setup-uv@v4
+        with:
+          version: "0.5.x"
+          enable-cache: true
+          cache-suffix: ${{ runner.os }}-${{ env.UV_PYTHON }}
+      - name: Install the project
+        run: uv sync --all-extras --dev
+      - name: Test with pytest
+        run: uv run --frozen pytest -q --junitxml=pytest.xml --cov=semantic_kernel --cov-report=term-missing:skip-covered ./tests/unit | tee python-coverage.txt
+      - name: Upload coverage
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: python-coverage-${{ env.FILE_ID }}.txt
+          path: python/python-coverage.txt
+          overwrite: true
+          retention-days: 1
+      - name: Upload pytest.xml
+        if: always()
+        uses: actions/upload-artifact@v4
         with:
-          github-token: ${{ secrets.GH_ACTIONS_PR_WRITE }}
-          run-id: ${{ github.event.workflow_run.id }}
-          path: python/
-          merge-multiple: true
-      - name: Display structure of downloaded files
-        run: ls python/
+          name: pytest-${{ env.FILE_ID }}.xml
+          path: python/pytest.xml
+          overwrite: true
+          retention-days: 1
       - name: Pytest coverage comment
         id: coverageComment
         uses: MishaKav/pytest-coverage-comment@main
-        continue-on-error: true
+        continue-on-error: false
```

Comment:
I've noticed over a few past PR iterations this bool flips from true -> false -> true -> false? Wondering why we keep changing it?

Reply:
Probably because this workflow was unstable. I noticed that it had been failing.